debug-and-error-logs.md

```diff
@@ -0,0 +1,115 @@
+
+### Severity Levels
+
+CockroachDB uses a sequence of severity levels for each message to help operators understand the likelihood they need to intercede:
+
+1. `INFO` *(lowest severity; no action necessary)*
+2. `WARNING`
+3. `ERROR`
+4. `FATAL` *(highest severity; requires operator attention)*
+
+**Default Behavior by Severity Level**
+
+Command | `INFO` messages | `WARNING` and above messages
+--------|--------|--------------------
+[`cockroach start`](start-a-node.html) | Logged to file | Logged to file &amp; `stderr`
+[All other commands](cockroach-commands.html) | Discarded | Logged to `stderr`
+
+### Output Locations
+
+Based on the command's flags and the message's severity level, CockroachDB does one of the following: 
+
+- [Writes the message to a file](#write-to-file)
+- [Prints it to `stderr`](#print-to-stderr)
+- [Discards it entirely](#discard-message)
+
+#### Write to File
+
+When file logging is enabled (by default for `cockroach start` or by including the `--log-dir` [flag](#flags) for other commands), CockroachDB writes messages to files in the directory specified by `--log-dir`. 
+
+The log's filenames use the following format:
+
+~~~
+cockroach.[host].[user].[start date/time + process ID].log
+~~~
+
+You can control the severity level of which messages are written to files with the `--log-file-verbosity` [flag](#flags).
+
+**Defaults per Command**
+
+Command | Log file location (`--log-dir`)
+--------|-------------|
+[`cockroach start`](start-a-node.html) | `--log-dir=<first `[`store`](start-a-node.html#store)` dir>/logs`<br/><br/>This location is also printed in the **Logs** field of the [standard output](start-a-node.html#standard-output) of `cockroach start`.
+[All other commands](cockroach-commands.html) | *N/A (Messages are not written to files. To log messages from commands besides `start`, set the `--log-dir` [flag](#flags) to a directory.)* |
+
+#### Print to `stderr`
+
+`stderr` is a standard Unix interface, which normally prints messages to a machine's console.
+
+By default, all commands print messages with a [severity level](#severity-levels) `WARNING` or greater to `stderr`. However, you can control the severity level of messages printed to `stderr` with the `--logtostderr` [flag](#flags).
+
+#### Discard Message
+
+Messages with severity levels below the `--logtostderr` and `--log-file-verbosity` values are neither written to files nor printed to `stderr`, so they are discarded.
+
+By default, commands besides `cockroach start` have their messages with the `INFO` [severity level](#severity-levels) totally discarded.
+
+## Flags
+
+{% include custom/logging-flags.md %}
+
+The `--log-backtrace-at` and `--verbosity` flags are intended for internal debugging by CockroachDB contributors.
+
+## Log Queries
```

Comment:
Not sure whether it should be mentioned here or in "see also" below, but to troubleshoot long-running query we would first recommend people to look at the virtual table `node_statement_statistics`, before they enable additional logging flags. That is always available and will not cause any performance overhead.